### PR TITLE
Fix Issue 21477 - TypeInfo errors in betterC are cryptic

### PR DESCRIPTION
**Pull request by WalterBright**
Had to add a parameter to the function.

### CODE REVIEW

File: src/dmd/typinf.d [hunk 1]
```diff
@@ -43,7 +45,10 @@ extern (C++) void genTypeInfo(const ref Loc loc, Type torig, Scope* sc)
     {
         if (!global.params.useTypeInfo)
         {
-            .error(loc, "`TypeInfo` cannot be used with -betterC");
+            if (e)
+                .error(loc, "expression `%s` uses the gc and cannot be used with -betterC", e.toChars());
```
Comment (ibuclaw) [inline src/dmd/typinf.d:49]: I'd suggest that GC should be capitalized here.

**Comment by WalterBright** [comment]
I thought for sure there'd already be a test case for this in the suite. Apparently not!

File: src/dmd/typinf.d (hunk 1 above)
Comment (WalterBright) [reply src/dmd/typinf.d]: ok

**Comment by WalterBright** [comment]
The buildkite errors in libmir seem to be irrelevant to this PR.

**Comment by thewilsonator** [comment]
Indeed it is.

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**